lib/spack/spack/build_environment.py

```diff
@@ -326,12 +322,32 @@ def set_build_environment_variables(pkg, env, dirty=False):
         env.set('SPACK_COMPILER_EXTRA_RPATHS', extra_rpaths)
 
     # Add bin directories from dependencies to the PATH for the build.
-    bin_dirs = reversed(
-        [d.prefix.bin for d in pkg.spec.dependencies(deptype='build')
-         if os.path.isdir(d.prefix.bin)])
-    bin_dirs = filter_system_bin_paths(bin_dirs)
-    for item in bin_dirs:
+    for prefix in build_prefixes:
```

Comment:
This grabs all the direct dependencies (not transitive), and it returns them in an arbitrary order. If it did return them in some order I'm not sure why that would be important here anyhow.